test/new_relic/agent/stats_engine_test.rb

```diff
@@ -178,22 +178,58 @@ def test_record_scoped_and_unscoped_metrics_records_multiple_unscoped_metrics
     ])
   end
 
-  def test_record_scoped_and_unscoped_metrics_is_thread_safe
+  def test_record_scoped_and_unscoped_metrics_is_thread_safe_without_traced_threads
```

Comment:
So i updated the failing test use disable `instrumentation.thread.tracing`, and then added a second version of this test below that has it enabled and moves the transaction outside. that one still fails, but I wanted to have both situations represented.